src/System.Private.ServiceModel/tools/test/SelfHostWcfService/WcfService.cs

```diff
@@ -367,6 +368,23 @@ public void ReturnContentType(string contentType)
             httpRespononseMessageProperty.Headers[HttpResponseHeader.ContentType] = contentType;
         }
 
+        public string EchoCookie()
+        {
+            string cookie = WebOperationContext.Current.IncomingRequest.Headers[HttpRequestHeader.Cookie];
+            return cookie;
+        }
+
+        public string EchoTimeAndSetCookie(string name)
+        {
+            string cookie = WebOperationContext.Current.IncomingRequest.Headers[HttpRequestHeader.Cookie];
+            string value = DateTime.Now.ToString();
+            cookie = $"{name}={value}";
+            WebOperationContext.Current.OutgoingResponse.Headers.Add($"Set-Cookie: {cookie};");
+            return value;
+        }
+
+        private static readonly string s_testCookieName = "test_cookie";
```

Comment:
I should have removed this variable. It was used in an early version of the test.